scio-test/src/it/scala/com/spotify/scio/benchmarks/ScioBenchmark.scala

```diff
@@ -148,74 +127,111 @@ final case class ScioBenchmarkLogger[F[_]](loggers: BenchmarkLogger[F]*) {
+  private val BatchMetrics = Set("Elapsed",
+                                 "TotalMemoryUsage",
+                                 "TotalPdUsage",
+                                 "TotalShuffleDataProcessed",
+                                 "TotalSsdUsage",
+                                 "TotalStreamingDataProcessed",
+                                 "TotalVcpuTime")
+  private val StreamingMetrics = Set(
+    "CurrentMemoryUsage",
+    "CurrentPdUsage",
+    "CurrentVcpuCount",
+    "TotalMemoryUsage",
+    "TotalPdUsage",
+    "TotalShuffleDataProcessed",
+    "TotalSsdUsage",
+    "TotalStreamingDataProcessed",
+    "TotalVcpuTime",
+    "SystemLag"
+  )
 
-  def batch(name: String, extraArgs: Array[String], scioResult: ScioResult): BenchmarkResult = {
+  def batch(timestamp: Instant,
+            name: String,
+            extraArgs: Array[String],
+            scioResult: ScioResult): BenchmarkResult = {
     require(scioResult.isCompleted)
 
     val job: Job = scioResult.as[DataflowResult].getJob
     val startTime: LocalDateTime = DateTimeParser.parseLocalDateTime(job.getCreateTime)
     val finishTime: LocalDateTime = DateTimeParser.parseLocalDateTime(job.getCurrentStateTime)
-    val elapsedTime: Seconds = Seconds.secondsBetween(startTime, finishTime)
+    val elapsedTime: Long = Seconds.secondsBetween(startTime, finishTime).getSeconds.toLong
 
-    val metrics: Map[String, String] = scioResult
+    val metrics: List[Metric] = Metric("Elapsed", elapsedTime.toString) :: scioResult
       .as[DataflowResult]
       .getJobMetrics
       .getMetrics
       .asScala
       .filter(metric => BatchMetrics.contains(metric.getName.getName))
-      .map(m => (m.getName.getName, m.getScalar.toString))
-      .sortBy(_._1)
-      .toMap
+      .map(m => Metric(m.getName.getName, m.getScalar.toString))
+      .toList
 
     BenchmarkResult(
+      timestamp,
       name,
       Some(elapsedTime),
       CircleCI.map(_.buildNum).getOrElse(-1L),
-      startTime,
-      Some(finishTime),
-      scioResult.state,
+      CircleCI.map(_.gitHash).getOrElse("none"),
+      job.getCreateTime,
+      Some(job.getCurrentStateTime),
+      scioResult.state.toString,
       extraArgs,
       metrics,
       BuildInfo.version,
       BuildInfo.beamVersion
     )
   }
 
-  def streaming(name: String,
+  def streaming(timestamp: Instant,
+                name: String,
                 buildNum: Long,
+                gitHash: String,
                 createTime: String,
                 jobMetrics: JobMetrics): BenchmarkResult = {
-    val startTime: LocalDateTime = DateTimeParser.parseLocalDateTime(createTime)
-
     val metrics = jobMetrics.getMetrics.asScala
       .filter(metric => StreamingMetrics.contains(metric.getName.getName))
-      .map(m => (m.getName.getName, m.getScalar.toString))
-      .sortBy(_._1)
-      .toMap
+      .map(m => Metric(m.getName.getName, m.getScalar.toString))
```

Comment:
maybe we should just represent `Metric.value` as a scalar, not a string? seems like there's a lot of conversions and the `compareMetrics` method assumes they can all be converted to Doubles anyway